src/Tasks/Microsoft.NET.Build.Tasks/CollectSDKReferencesDesignTime.cs

```diff
@@ -34,15 +34,36 @@ public class CollectSDKReferencesDesignTime : TaskBase
 
         protected override void ExecuteCore()
         {
-            ImplicitPackageReferences = 
-                PreprocessPackageDependenciesDesignTime.GetImplicitPackageReferences(DefaultImplicitPackages);
+            ImplicitPackageReferences = GetImplicitPackageReferences(DefaultImplicitPackages);
             
             var sdkDesignTimeList = new List<ITaskItem>(SdkReferences);
             sdkDesignTimeList.AddRange(GetImplicitPackageReferences());
 
             SDKReferencesDesignTime = sdkDesignTimeList.ToArray();
         }
 
+        internal static HashSet<string> GetImplicitPackageReferences(string defaultImplicitPackages)
```

Comment:
This method was also copied into `ResolvePackageAssets`.  Can you unify these?

Reply:
done